### PR TITLE
Fixes atmos connector continuity during shuttle transit

### LINKED ISSUE #18050: canisters disconnect from connector ports when travelling in shuttles
(WEB REPORT BY: spoonylove REMOTE: 172.93.109.202:7777)
# Revision
c8e35f23bfa1edbdb3f7797fbbc8cb0d20baf214

# Description
experienced on metaclub vox shuttle, connected canister to connector for planned borer farm, luckily empty. when I took off it disconnected, had it been open I wouldve flooded the ship with plasma.

# Steps to Reproduce
build connector port, attach canister, take off, watch it disconnect on take off

# What you Expected
not that

# What Actually Happened
a deadly accident had I not test ran the thing.

### PR DESCRIPTION
**Pull request by Pathid**
Closes #18050

:cl:
* bugfix: Portable atmos devices hooked to connectors aboard shuttles will no longer disconnect on departure and arrival.

### CODE REVIEW

**Comment by DamianX** [comment]
There must be a better way

**Comment by Pathid** [comment]
Probably.

@DamianX I suppose I could add a new argument to forceMove()?

File: code/ATMOSPHERICS/components/unary/portables_connector.dm [hunk 1]
```diff
@@ -47,7 +47,9 @@
 	if(!connected_device)
 		return
 	if(AM == connected_device)
-		connected_device.disconnect()
+		spawn(1)
+			if(AM.loc != src.loc)
```
Comment (DamianX) [inline code/ATMOSPHERICS/components/unary/portables_connector.dm]: AM may be null now
src may be null now
connected_device may be null now

**Comment by DamianX** [comment]
Yeah, that's what I was thinking, if you feel like it.

**Comment by PJB3005** [comment]
Could also move it to atom locking and just tell dimensional push to fuck off.

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
There has to be a better way

**Comment by DamianX** [comment]
better way when?

**Pull request closed without merging**